### PR TITLE
lateDependsOn => plugins.withType(JavaPlugin)

### PR DESCRIPTION
**Pull request by brunobowden**
- plugins.withType(JavaPlugin) to delay adding dependencies
- “buildAllObjcLibraries” => “j2objcBuildAllObjcLibraries”

### CODE REVIEW

**Comment by advayDev1** [comment]
hmm won't this reintroduce the issue i'm trying to fix with #157 ?

**Comment by brunobowden** [comment]
@confile - I think this is a better way of deferring the dependsOn for the JavaPlugin. From testing, it nolonger depends on the 'java' and 'j2objc' apply ordering. When the 'java' plugin isn't specified, then it correctly fails but it's not such a great error message:

```
Brunos-Air:taptap brunobowden$ ./gradlew clean base:j2objcXcode
Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.
org.gradle.api.InvalidUserDataException: j2objc plugin didn't find the 'java' plugin in the 'base' project.
This is a requirement for using j2objc. Please see usage information at:

https://github.com/j2objc-contrib/j2objc-gradle/#usage
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	...
```

**Comment by advayDev1** [comment]
@brunobowden  - it solves only the plugin ordering issue, but not the issue in #157.  as soon as the java plugin is applied (right after the j2objc plugin), then tasks.getByName will be called, assemble task will be created and we won't be able to add native binaries.

**Comment by brunobowden** [comment]
It defers evaluation until the JavaPlugin is loaded... so the tasks exist
by that point. I tested it by reordering the plugins to load the 'j2objc'
plugin first and then the 'java' plugin and it works fine.

**Comment by advayDev1** [comment]
yes agreed, but we still can't add native binaries after assemble is created.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPlugin.groovy [hunk 1]
```diff
@@ -27,6 +27,9 @@ import org.gradle.api.InvalidUserDataException
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.logging.LogLevel
+import org.gradle.api.plugins.BasePlugin
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPlugin.groovy]: unused import

**Comment by brunobowden** [comment]
Deferring to @advayDev1 fix but we should still look in further to see if there's a better way to do this.

**Comment by advayDev1** [comment]
Sorry I did not mean you should discard the entire PR.  The clean up of JavaPlugin vs. ours ordering wise still works after my PR #157 (which is merged), and I believe is still a valuable cleanup.  Just revert the part of the PR that touches lateDependsOn.

**Comment by brunobowden** [comment]
Rename to `j2objcBuildAllObjcLibraries` was done in commit 7f53fc7.

**Pull request closed without merging**